libsolidity/codegen/ABIFunctions.cpp

```diff
@@ -1393,7 +1391,7 @@ string ABIFunctions::abiDecodingFunctionCalldataArray(ArrayType const& _type)
 		w("functionName", functionName);
 		w("readableTypeName", _type.toString(true));
 		w("baseEncodedSize", toCompactHexWithPrefix(_type.isByteArray() ? 1 : _type.baseType()->calldataEncodedSize()));
```

Comment:
Please add an assertion that `baseType()->calldataEncodedSize()` is not zero.

Reply:
Actually the assertion ``_type.baseType()->calldataEncodedSize() < u256("0xffffffffffffffff")`` above is trivially true, since ``calldataEncodedSize()`` is of type ``unsigned int``, so I'll just change that one (or do you think we should keep it in case we change the type of ``calldataEncodedSize()``? Probably not - calldata is never that large anyways)

Reply:
Ah no - nevermind, I miscounted the ``f``s.

Reply:
Hm... for the emscripten build ``unsigned int`` will be 32-bit, though... we should probably think about whether that's a problem (I was wondering about whether that's an issue somewhere in the code before)...

Reply:
Me might want to consider removing all `unsigned int` types.